internal/web/server.go

```diff
@@ -468,6 +498,49 @@ func defaultKubeconfigExists() bool {
 	}
 }
 
+func (s *server) initInformer() (cache.Store, cache.Controller) {
+	return cache.NewInformer(
+		&cache.ListWatch{
+			ListFunc: func(options metav1.ListOptions) (runtime.Object, error) {
+				var pkgList v1alpha1.PackageList
+				err := s.pkgClient.Packages().GetAll(context.Background(), &pkgList)
+				return &pkgList, err
+			},
+			WatchFunc: func(options metav1.ListOptions) (watch.Interface, error) {
+				return s.pkgClient.Packages().Watch(context.Background())
+			},
+		},
+		&v1alpha1.Package{},
+		0,
+		cache.ResourceEventHandlerFuncs{
+			AddFunc: func(obj interface{}) {
+				if pkg, ok := obj.(*v1alpha1.Package); ok {
+					s.broadcastPkg(pkg, client.GetStatusOrPending(&pkg.Status), nil, "")
+				}
+			},
+			UpdateFunc: func(oldObj, newObj interface{}) {
+				if pkg, ok := newObj.(*v1alpha1.Package); ok {
+					ctx := client.SetupContextWithClient(context.TODO(), s.restConfig, s.rawConfig, s.pkgClient)
+					if mf, err := manifest.GetInstalledManifestForPackage(ctx, *pkg); err != nil && !errors.Is(err, manifest.ErrPackageNoManifest) {
+						fmt.Fprintf(os.Stderr, "Error fetching manifest for package %v: %v\n", pkg.Name, err)
+					} else {
+						var idx repo.PackageIndex
```

Comment:
Since the `latestVersion` is now in the repo index as well, I think it would make sense to download that once and then keep it in memory. That way, we don't have to make a new request to the repo every time a package is updated (which can happen quite frequently).

Reply:
I implemented a mini-cache which holds the index for 5 minutes now, but I don't like it one bit and I think we should migrate it to a proper cache for all the repo methods. What do you think?

Reply:
I agree that the current implementation seems a little half baked. What do you have in mind regarding a "proper" cache?

Reply:
I meant to say that we should consider caching at other places too (all sorts of manifest and repo index fetching), probably also in the operator? 
Something like this library: https://github.com/eko/gocache could help.

Reply:
I will create a ticket, I think its out of scope for this one.